tasks/managed/embargo-check/embargo-check.yaml

```diff
@@ -62,19 +71,58 @@ spec:
+        set -x
+        CVE_FIELD="$(jq -r '.[] | select(.name=="CVE ID") | .id' <<< "$CUSTOM_RH_FIELDS")"
+        COMPONENT_FIELD="$(jq -r '.[] | select(.name=="Downstream Component Name") | .id' <<< "$CUSTOM_RH_FIELDS")"
+
         RC=0
 
         NUM_ISSUES=$(jq -cr '.releaseNotes.issues.fixed | length' "${DATA_FILE}")
         for ((i = 0; i < NUM_ISSUES; i++)); do
             issue=$(jq -c --argjson i "$i" '.releaseNotes.issues.fixed[$i]' "${DATA_FILE}")
             server=$(jq -r '.source' <<< "$issue")
             API=$(jq -r '.[] | select(.servers[] | contains("'"$server"'")) | .api' <<< "$SUPPORTED_ISSUE_TRACKERS")
-            API_URL="https://$(jq -r '.source' <<< "$issue")/${API}/$(jq -r '.id' <<< "$issue")"
-            curl --fail --output /dev/null "${API_URL}"
+            API_URL="https://"
+            # Add authentication for issues.redhat.com
+            if [ "$server" = "issues.redhat.com" ] ; then
+                API_URL="-H \"Authorization: Bearer $ACCESS_TOKEN\" ${API_URL}"
+            fi
+            API_URL="${API_URL}$(jq -r '.source' <<< "$issue")/${API}/$(jq -r '.id' <<< "$issue")"
+            OUTPUT=$(curl --retry 3 --fail "${API_URL}")
             if [ $? -ne 0 ] ; then
-                echo "${issue} is not publicly visible. Assuming it is embargoed and stopping pipelineRun execution."
+                echo "${issue} is not visible. Assuming it is embargoed and stopping pipelineRun execution."
                 RC=1
+                continue
+            fi
+
+            # Perform additional checks only for Vulnerability issues for issues.redhat.com
+            if [ "$server" != "issues.redhat.com" ] ; then
+                continue
+            fi
+            ISSUE_TYPE=$(jq -r '.fields.issuetype.name' <<< "$OUTPUT")
+            if [ "$ISSUE_TYPE" != "Vulnerability" ] ; then
+                continue
+            fi
+            CVE_ID=$(jq -r --arg field "$CVE_FIELD" '.fields[$field]' <<< "$OUTPUT")
+            COMPONENT_NAME=$(jq -r --arg field "$COMPONENT_FIELD" '.fields[$field]' <<< "$OUTPUT")
+            if ! COMPONENT_IN_RELEASE_NOTES="$(jq -ec --arg name "$COMPONENT_NAME" \
+              '.releaseNotes.content.images[] | select(.component==$name)' "$DATA_FILE")"; then
+                echo -n "Issue $issue lists 'Downstream Component Name' $COMPONENT_NAME but that component does"
+                echo " not appear in releaseNotes.content.images. Failing"
+                RC=2
+            fi
+
+            if [ "$(jq --arg CVE "$CVE_ID" '.cves.fixed | has($CVE)' <<< "$COMPONENT_IN_RELEASE_NOTES")" != "true" ]
+            then
+                echo -n "Issue $issue lists 'Downstream Component Name' $COMPONENT_NAME and 'CVE ID' $CVE_ID"
+                echo " but that CVE is not present in the releaseNotes.content.images section for that component."
+                echo "Failing"
```

Comment:
I think saying "Failing" made would make more sense if we were exiting here. But we only fail after the loop (so that we could collect and print all the failures), so maybe it would make more sense to add an "Error: " prefix to the messages and remove "Failing"? (This applies elsewhere too.)

Reply:
Updated